autotx/utils/ethereum/helpers/swap_from_eoa.py

```diff
@@ -16,19 +17,24 @@ def swap(
 ) -> None:
     txs = build_swap_transaction(
         client,
-        Decimal(amount),
+        Decimal(str(amount)),
         from_token,
         to_token,
         ETHAddress(user.address),
         False,
         chain
     )
 
-    for i, tx in enumerate(txs):
+    for tx in txs:
+        if chain in [ChainId.GNOSIS, ChainId.BASE_MAINNET]:
```

Comment:
Any more background on why this is needed?

Reply:
sometimes the gas estimation received from li.fi in these two chains is not correct, making the swap from an eoa to fail - it's different when it comes to executing from safe because we calculate the gas everytime we execute transaction using safe

Reply:
Maybe we should always estimate the gas then? Who knows if another chain might fail or li.fi changes something and causes previously working ones to sometimes fail

Reply:
sure - no strong opinion there, will do that :+1: